classes/rest/endpoints/RestEndpointTransfer.class.php

```diff
@@ -350,12 +350,11 @@ public function post($id = null, $add = null) {
             // Build options from provided data and defaults
             $options = array();
             foreach(Transfer::allOptions() as $name => $dfn)  {
-                $value = $dfn['default'];
-                
-                if(in_array($name, $allowed_options) && $data->options->exists($name))
-                    $value = $data->options->$name;
-                
-                $options[$name] = $value;
+                if(in_array($name, $allowed_options)
+                    && $data->options->exists($name))
+                {
+                    $options[$name] = $data->options->$name;
+                }
             }
             
```

Comment:
code below here uses $options['key'] for example using key=TransferOptions::GET_A_LINK. There are only a handful of these accesses but they will not see any default. While what is there might work ok now, it might be brittle to options which are not directly set (and want a default) in the future. 

Maybe we should set the options early, after the for loop;
```
$transfer->options = $options;
```
and then convert those direct $options['key'] lookups to use $transfer->getOption('key').

This update could be done in a subsequent PR, so it's not a blocker :)